vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -116,6 +116,14 @@ def _filter_interaction(
 
     @_log_call
     def build(self):
+        clientside_callback(
+            ClientsideFunction(namespace="clientside", function_name="update_graph_theme"),
+            # Output here to ensure that the callback is only triggered if the graph exists on the currently open page.
+            output=[Output(self.id, "figure")],
+            inputs=[Input("theme_selector", "checked"), State("vizro_themes", "data"), State(self.id, "id")],
```

Comment:
Can we instead do this by passing `State(self.id, "figure")` and then no need for the `document    .getElementById(graph_id)` in the callback function?

It would also be really nice if it were possible to actually use the `Output(self.id, "figure")` rather than needing to do a `no_update` - would that be possible if we passed in the graph itself as `State` rather than  just the id?

Reply:
Ideally, we should only affect the `layout` of the figure (not the `data` of the figure). One reason is performance (if the chart displays large data) and the other is that we don't want to redraw the entire chart because we want to allow the chart theme to change properly even when the chart is in the middle of an animation (e.g. on an animated gapminder tutorial). And, because `Patch` object is not available in the clientside_callbacks, I handled it using the `Plotly.relayout`.

Reply:
Cool, that makes sense and is what I suspected.

Is it possible to rewrite the `State(self.id, "id")` to use `State(self.id, "figure")` directly instead of doing the `document.getElementById(graph_id)`? No problem if not, but if it works then you could add it to #536

Reply:
I don't think this could work. See the `Plotly.relayout` documentation -> https://plotly.com/javascript/plotlyjs-function-reference/#plotlyrelayout. Its first argument has to be or DOM node or its ID (and the figure is just a JSON obj sent from the server). So, as we don't have the ID for this DOM node, we have to fetch the entire DOM node with the:
```js
  const plotly_graph = document
    .getElementById(graph_id)
    .querySelector(".js-plotly-plot");
```